terraform/main.tf

```diff
@@ -2,14 +2,97 @@ provider "aws" {
   region = "eu-central-1"  # Replace with your desired AWS region
 }
 
+# Random string generator for DNS names
+resource "random_string" "dns_name" {
+  count   = var.node_count  # Create a random string for each instance
+  length  = 6
```

Comment:
Can we limit this length to 3?
That should be largely sufficient to avoid name collisions, and safe us some typing.

Reply:
Yes. I've reduced it to 3 in new commit.
Also reduced the cluster dns name length 8 to 3.